ktlint-ruleset-experimental/src/main/kotlin/com/pinterest/ktlint/ruleset/experimental/ArgumentListWrappingRule.kt

```diff
@@ -193,4 +198,10 @@ class ArgumentListWrappingRule : Rule("argument-list-wrapping") {
             .firstOrNull { it.elementType == ElementType.TYPE_PARAMETER_LIST }
             ?.children()
             ?.any { it.isWhiteSpaceWithNewline() } == true
+
+    private fun ASTNode.isOnSameLineAsIfKeyword(): Boolean {
+        val containerNode = psi.getStrictParentOfType<KtContainerNode>() ?: return false
+        val ifExpression = containerNode.parent as? KtIfExpression ?: return false
```

Comment:
I have a feeling that `KtIfExpression` is not enough here, we should probably also consider `when` and `while`.

Reply:
Fixed in 74656c3